calyx-opt/src/passes/static_promotion.rs

```diff
@@ -428,73 +444,104 @@ impl Visitor for InferStaticTiming {
                     return Err(Error::malformed_structure(msg)
                         .with_pos(&grp.attributes));
                 }
-            } else {
-                drop(grp);
-                group
-                    .borrow_mut()
-                    .attributes
-                    .insert(ir::NumAttr::Static, latency);
             }
+            latency_result = Some(latency);
+        } else {
+            latency_result = None;
         }
 
-        // Compute the latency of the control-program.
-        let comp_lat: HashMap<ir::Id, u64> = self
-            .latency_data
-            .iter()
-            .filter_map(|(comp, go_done)| {
-                if go_done.ports.len() == 1 {
-                    Some((*comp, go_done.ports[0].2))
-                } else {
-                    None
+        if let Some(res) = latency_result {
+            builder
+                .component
+                .get_groups_mut()
+                .remove(s.group.borrow().name());
+            let sg = builder.add_static_group(s.group.borrow().name(), res);
+            for assignment in s.group.borrow().assignments.iter() {
+                if !(assignment.dst.borrow().is_hole()
+                    && assignment.dst.borrow().name == "done")
+                {
+                    let static_s = Assignment::from(assignment);
+                    sg.borrow_mut().assignments.push(static_s);
                 }
-            })
-            .collect();
-
-        if let Some(time) = comp.control.borrow_mut().update_static(&comp_lat) {
-            let mut go_ports = comp
-                .signature
-                .borrow()
-                .find_all_with_attr(ir::NumAttr::Go)
-                .collect_vec();
+            }
+            let s_enable = ir::StaticControl::Enable(ir::StaticEnable {
+                group: Rc::clone(&sg),
+                attributes: s.attributes.clone(),
```

Comment:
Seems like this clone of `s.attributes` would be unnecessary since `s` will be removed?